lib/src/view/user/full_games_screen.dart

```diff
@@ -0,0 +1,277 @@
+        debugPrint(
+          'SEVERE: [FullGames] could not load game history; $error\n$stackTrace',
+        );
+        return Padding(
+          padding: Styles.bodySectionPadding,
+          child: const Text('Could not load game history.'),
+        );
+      },
+      loading: () => const Center(child: CircularProgressIndicator.adaptive()),
+    );
+  }
+}
+
+class _GameList extends ConsumerStatefulWidget {
+  const _GameList({this.userId, required this.initialPage});
+  final UserId? userId;
+  final FullGamePaginator initialPage;
+
+  @override
+  _GameListState createState() => _GameListState();
+}
+
+class _GameListState extends ConsumerState<_GameList> {
+  ScrollController controller = ScrollController();
+
+  late List<LightArchivedGame> games;
+  late FullGamePaginator page;
+  late UserId? userId;
+  bool isLoading = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    userId = widget.userId;
+    page = widget.initialPage;
+    games = page.games.toList();
+    controller = ScrollController()..addListener(_scrollListener);
+  }
+
+  @override
+  void dispose() {
+    controller.removeListener(_scrollListener);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Scrollbar(
+        child: ListView.builder(
+          controller: controller,
+          itemBuilder: (context, index) {
+            if (index == games.length) {
+              if (isLoading || page.currentPage == 1) {
+                return const Center(
+                  heightFactor: 2.0,
+                  child: CircularProgressIndicator.adaptive(),
+                );
+              } else {
+                return kEmptyWidget;
+              }
+            } else {
+              return _SlideMenu(
+                menuItems: <Widget>[
+                  IconButton(
+                    icon: const Icon(Icons.bookmark_add_outlined),
+                    onPressed: () {},
+                  ),
+                ],
+                child: ExtendedGameListTile(
+                  game: games[index],
+                  userId: userId,
+                ),
+              );
+            }
+          },
+          itemCount: games.length + 1,
+        ),
+      ),
+    );
+  }
+
+  Future<void> _scrollListener() async {
+    if (controller.position.extentAfter < 500 &&
+        page.nextPage != null &&
+        !isLoading) {
+      isLoading = true;
+      final nextPage = (userId != null
+          ? await ref.read(
+              _userFullGamesProvider(
+                page: page.nextPage!,
+                userId: userId!,
+              ).future,
+            )
+          : await ref.read(accountFullGamesProvider(page.nextPage!).future));
+      if (nextPage != null) page = nextPage;
+
+      setState(() {
+        games.addAll(nextPage?.games ?? []);
+      });
+      isLoading = false;
+    }
+  }
+}
+
+class _SlideMenu extends StatefulWidget {
```

Comment:
I'm not against a custom implementation, but is there a reason you choose it instead of `flutter_slidable` package which is already used in the project?

Reply:
Oh I just didn't know it was already in the project 🤣